### PR TITLE
#0: Add basic sanity checks during matmul program config creation

### PR DESCRIPTION
**Pull request by TT-BrianLiu**
~~- Add MatmulBaseProgramConfig which does basic sanity checks on:~~
~~* compute_with_storage_grid_size~~
~~* in0_block_w~~
~~* out_subblock_h~~
~~* out_subblock_w~~
~~* per_core_M~~
~~* per_core_N~~
~~- Remove reference to designated initializers for all matmul program config~~

EDIT: Going with the simpler approach of just adding it to matmul op so the structs don't need to be modified.

### CODE REVIEW

**Review by bbradelTT — APPROVED** [review]

File: tt_eager/tt_dnn/op_library/bmm/bmm_op.cpp [hunk 1]
```diff
@@ -331,15 +331,16 @@ tt::operations::primary::MatmulMultiCoreReuseMultiCast1DProgramConfig get_mcast_
...
-        .compute_with_storage_grid_size = grid_size,
-        .in0_block_w = in0_block_w,
-        .out_subblock_h = out_subblock_h,
-        .out_subblock_w = out_subblock_w,
-        .per_core_M = per_core_M,
-        .per_core_N = per_core_N,
-        .fuse_batch = fuse_batch,
-        .fused_activation = fused_activation,
-        .mcast_in0 = mcast_in0};
+        grid_size,
```
Comment (xanderchin) [inline tt_eager/tt_dnn/op_library/bmm/bmm_op.cpp:334]: Is there a reason why we are removing named struct memebers?

File: tests/tt_eager/integration_tests/test_bert.cpp [hunk 2]
```diff
@@ -205,30 +200,87 @@ void test_bert() {
...
-        parameters.emplace(fmt::format("ff1_bias_{}", encoder_index), tt::numpy::random::uniform(bfloat16(-1.0f), bfloat16(1.0f), {1, 1, TILE_HEIGHT, intermediate_size}, Layout::TILE).to(device, dram_memory_config));
-        parameters.emplace(fmt::format("ff2_weight_{}", encoder_index), tt::numpy::random::uniform(bfloat16(-1.0f), bfloat16(1.0f), {1, 1, intermediate_size, hidden_size}, Layout::TILE).to(device, dram_memory_config));
-        parameters.emplace(fmt::format("ff2_bias_{}", encoder_index), tt::numpy::random::uniform(bfloat16(-1.0f), bfloat16(1.0f), {1, 1, TILE_HEIGHT, hidden_size}, Layout::TILE).to(device, dram_memory_config));
-        parameters.emplace(fmt::format("feedforward_layernorm_weight_{}", encoder_index), tt::numpy::random::uniform(bfloat16(-1.0f), bfloat16(1.0f), {1, 1, TILE_HEIGHT, TILE_WIDTH}, Layout::ROW_MAJOR).to(device, dram_memory_config));
-        parameters.emplace(fmt::format("feedforward_layernorm_bias_{}", encoder_index), tt::numpy::random::uniform(bfloat16(-1.0f), bfloat16(1.0f), {1, 1, TILE_HEIGHT, TILE_WIDTH}, Layout::ROW_MAJOR).to(device, dram_memory_config));
+        parameters.emplace(
```
Comment (xanderchin) [inline tests/tt_eager/integration_tests/test_bert.cpp:210]: Maybe clang format off for this?

File: tt_eager/tt_dnn/op_library/bmm/bmm_op.cpp (hunk 1 above)
Comment (TT-BrianLiu) [reply tt_eager/tt_dnn/op_library/bmm/bmm_op.cpp:334]: Cause it's only allowed if we use default constructor

File: tests/tt_eager/integration_tests/test_bert.cpp (hunk 2 above)
Comment (TT-BrianLiu) [reply tests/tt_eager/integration_tests/test_bert.cpp:210]: I will revert all the clang format.

**Comment by TT-BrianLiu** [comment]
Passing pipelines:
- post-commit: https://github.com/tenstorrent/tt-metal/actions/runs/9274110326
- ttnn unit tests: https://github.com/tenstorrent/tt-metal/actions/runs/9274141625
- ttnn sweep tests (properly times out now, instead of crashing): https://github.com/tenstorrent/tt-metal/actions/runs/9274134834/job/25516001387

**Pull request merged**